wasm.md

```diff
@@ -636,6 +650,104 @@ Unlike labels, only one frame can be "broken" through at a time.
     rule <k> (return => .) ~> FR:Frame ... </k>
 ```
 
+Memory
+------
+
+When memory is allocated, it is put into the store at the next available index.
+Memory can only grow in size, so the minimum size is the initial value.
+
+```k
+    syntax Instr ::= "(" "memory"               ")"
+                   | "(" "memory" Int           ")" // Max only
+                   | "(" "memory" Int Int       ")" // Min and max.
+                   |     "memory" "{" Int MemBound "}"
+ // --------------------------------------------------
+    rule <k> ( memory                 ) => memory { 0   .MemBound } ... </k>
+    rule <k> ( memory         MAX:Int ) => memory { 0   MAX       } ... </k>
+    rule <k> ( memory MIN:Int MAX:Int ) => memory { MIN MAX       } ... </k>
+
+    rule <k> memory { _ _ } => trap ... </k>
+         <memAddrs> MAP </memAddrs> requires MAP =/=K .Map
+
+    rule <k> memory { MIN MAX } => . ... </k>
+         <memAddrs>    .Map => (0 |-> NEXT)  </memAddrs>
+         <nextMemAddr> NEXT => NEXT +Int 1 </nextMemAddr>
+         <mems>
+           ( .Bag
+          => <memInst>
+               <memAddr> NEXT </memAddr>
+               <mmax>    MAX  </mmax>
+               <msize>   MIN  </msize>
+             </memInst>
+           )
+           ...
+         </mems>
+```
+TODO: There are many more valid ways to instantiate memory. Allow specifying just max or neither min or max, folded syntax, identifier, inline instantiation, and inline export and import.
```

Comment:
Make the `TODO` bold with `**` around it (eg. `**TODO**`). Also, move above the code block and make sure there is a blank line between the code-block and the TODO.

One reason for such pickiness about formatting is the line-numbers reported as errors at `kompile` time make more sense when we make sure that the `pandoc`-emitted code will line up line-for-line with the original.

Reply:
I have noticed some wonkiness with the line numbers in the error messages, good to know what the cause is.